### PR TITLE
Update HEAD to support Bazel 6 -> Bazel 8.

### PR DESCRIPTION
**Pull request by jin**
..now that Bazel 8 is released.

This is still mostly blocked by a Bazel 8 compatible rules_android release.

To be precise on what is supported:

* rules_jvm_external can be used by Bazel 8 projects via bzlmod, except for when the projects use rules_android, rules_scala and other dependencies that do not yet support Bazel 8. See the bzlmod example for a working example, which is already on Bazel 8. Other examples are pinned to Bazel 7.4.1 while we upgrade them incrementally.
* rules_jvm_external's own .bazelversion is still on 7.4.1, until we resolve the rules_android blockers.

### CODE REVIEW

**Comment by jin** [comment]
Blockers:

https://buildkite.com/bazel/rules-jvm-external/builds/4695#0193ae64-8047-43f3-8481-e26593c7f6a9

```
ERROR: Traceback (most recent call last):
--
  | File "/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/ec321eb2cc2d0f8f91b676b6d4c66c29/external/+maven+regression_testing_coursier/BUILD", line 48, column 11, in <toplevel>
  | aar_import(
  | Error in aar_import: Couldn't auto load 'aar_import' from '@rules_android//rules:rules.bzl'.
  | WARNING: Target pattern parsing failed.
  | ERROR: Skipping '@regression_testing_coursier//:pin': no such target '@@+maven+regression_testing_coursier//:pin': target 'pin' not declared in package '' defined by /var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/ec321eb2cc2d0f8f91b676b6d4c66c29/external/+maven+regression_testing_coursier/BUILD
  | ERROR: no such target '@@+maven+regression_testing_coursier//:pin': target 'pin' not declared in package '' defined by /var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/ec321eb2cc2d0f8f91b676b6d4c66c29/external/+maven+regression_testing_coursier/BUILD
```

https://buildkite.com/bazel/rules-jvm-external-examples/builds/4462#0193ae64-9a66-49c9-a6c0-78754bf63d6b

```
(02:27:48) WARNING: Couldn't auto load rules or symbols, because no dependency on module/repository 'rules_android' found. This will result in a failure if there's a reference to those rules or symbols.
--
  | (02:27:49) ERROR: Traceback (most recent call last):
  | File "/workdir/examples/simple/BUILD", line 1, column 16, in <toplevel>
  | android_library(
  | Error in android_library: Couldn't auto load 'android_library' from '@rules_android//rules:rules.bzl'.
  | (02:27:49) ERROR: package contains errors:
  | (02:27:49) ERROR: package contains errors: : Traceback (most recent call last):
  | File "/workdir/examples/simple/BUILD", line 1, column 16, in <toplevel>
  | android_library(
  | Error in android_library: Couldn't auto load 'android_library' from '@rules_android//rules:rules.bzl'.
  | (02:27:49) WARNING: Target pattern parsing failed.
  | (02:27:49) ERROR: Skipping '//...': Error evaluating '//...': error loading package '': Package '' contains errors
  | (02:27:49) ERROR: Error evaluating '//...': error loading package '': Package '' contains errors
  | (02:27:49) INFO: Elapsed time: 1.869s
  | (02:27:49) INFO: 0 processes.
  | (02:27:49) ERROR: Build did NOT complete successfully
  | bazel build failed with exit code 1
```

https://buildkite.com/bazel/rules-jvm-external-examples/builds/4462#0193ae64-9a6a-4c4d-abc2-e850857d48eb

```
(02:27:45) WARNING: Couldn't auto load rules or symbols, because no dependency on module/repository 'rules_android' found. This will result in a failure if there's a reference to those rules or symbols.
--
  | (02:27:47) WARNING: Target pattern parsing failed.
  | (02:27:47) ERROR: Skipping '//...': error loading package under directory '': error loading package '': Unable to find package for @@[unknown repo 'bazel_common' requested from @@]//tools/maven:pom_file.bzl: The repository '@@[unknown repo 'bazel_common' requested from @@]' could not be resolved: No repository visible as '@bazel_common' from main repository. Was the repository introduced in WORKSPACE? The WORKSPACE file is disabled by default in Bazel 8 (late 2024) and will be removed in Bazel 9 (late 2025), please migrate to Bzlmod. See https://bazel.build/external/migration.
  | (02:27:47) ERROR: error loading package under directory '': error loading package '': Unable to find package for @@[unknown repo 'bazel_common' requested from @@]//tools/maven:pom_file.bzl: The repository '@@[unknown repo 'bazel_common' requested from @@]' could not be resolved: No repository visible as '@bazel_common' from main repository. Was the repository introduced in WORKSPACE? The WORKSPACE file is disabled by default in Bazel 8 (late 2024) and will be removed in Bazel 9 (late 2025), please migrate to Bzlmod. See https://bazel.build/external/migration.
```


https://buildkite.com/bazel/rules-jvm-external-examples/builds/4462#0193ae64-9a70-4884-8485-1213e820e258

```
(02:27:47) WARNING: Target pattern parsing failed.
--
  | (02:27:47) ERROR: Skipping '//...': error loading package under directory '': error loading package 'src/main': Unable to find package for @@[unknown repo 'rules_jvm_external' requested from @@]//:defs.bzl: The repository '@@[unknown repo 'rules_jvm_external' requested from @@]' could not be resolved: No repository visible as '@rules_jvm_external' from main repository. Was the repository introduced in WORKSPACE? The WORKSPACE file is disabled by default in Bazel 8 (late 2024) and will be removed in Bazel 9 (late 2025), please migrate to Bzlmod. See https://bazel.build/external/migration.
  | (02:27:47) ERROR: error loading package under directory '': error loading package 'src/main': Unable to find package for @@[unknown repo 'rules_jvm_external' requested from @@]//:defs.bzl: The repository '@@[unknown repo 'rules_jvm_external' requested from @@]' could not be resolved: No repository visible as '@rules_jvm_external' from main repository. Was the repository introduced in WORKSPACE? The WORKSPACE file is disabled by default in Bazel 8 (late 2024) and will be removed in Bazel 9 (late 2025), please migrate to Bzlmod. See https://bazel.build/external/migration.
```

(and many more similar ones from almost every example project)

**Comment by jin** [comment]
@ahumesky for the rules_android errors, we probably need to update that first

**Comment by jin** [comment]
I've pinned the `bzlmod` example and the main repo itself to Bazel 8. I'm also temporarily pinning all other failing examples to 7.4.1 while we migrate each of them gradually.

File: README.md [hunk 1]
```diff
@@ -92,11 +92,14 @@ Find other GitHub projects using `rules_jvm_external`
...
+* Bazel 6.4.0, up to the current LTS version.
+* Support for Bazel versions between `5.4` and `7.x` is only available on releases `6.x`.
+* Support for Bazel versions between `4.x` and `5.4` is only available on releases `5.x`.
+* Support for Bazel versions before `4.0.0` is only available on releases `4.2` or earlier.
 
-Support for Bazel versions between `4.x` and `5.4` is only available on rules_jvm_external releases `5.x`. 
-
-Support for Bazel versions before `4.0.0` is only available on rules_jvm_external releases `4.2` or earlier.
+**Compatibility guideline:** This project aims to be backwards compatible with
+the (current LTS - 2) version. If the current LTS version is 5, then we aim to
```
Comment (shs96c) [inline README.md:101]: Might as well use the current LTS of `8` here....

**Review by shs96c — APPROVED** [review]
LGTM. Thank you for doing this!

**Comment by jin** [comment]
Updated the PR description on exactly what works with Bazel 8 and not.

As expected, only the Bazel 8 job for rules_jvm_external's tests is failing now, because of `Error in aar_import: Couldn't auto load 'aar_import' from '@rules_android//rules:rules.bzl'.`, which is being worked on in https://github.com/bazel-contrib/rules_jvm_external/pull/1215. Everything else is green.

**Pull request merged**